### PR TITLE
Add improved Utilities codes

### PR DESCRIPTION
**Pull request by hagertnl**
Added unique states for NODE_FAIL, TIMEOUT, and FAIL, and changed the check status values to not conflict with standard reporting values

### CODE REVIEW

File: harness/utilities/update_runs_from_slurm.py [hunk 1]
```diff
@@ -84,13 +84,16 @@
...
 
 # Global parameters extracted for ease of use ##################################
-failed_job_event_name = 'check_end'  # Event name to log under for a failed job
-failed_job_event_value = '2'   # This is the status logged for the event
-success_job_event_name = 'check_end'  # Event name to log under for a completed job
-success_job_event_value = '0'   # This is the status logged for the completed event
-
+state_to_value = {
+    'fail': 21,
+    'timeout': 21,
```
Comment (verolero86) [inline harness/utilities/update_runs_from_slurm.py]: Update `timeout` code

**Comment by hagertnl** [comment]
Converted to draft - need to implement better error checking and a few more enhancements to the script.

**Comment by hagertnl** [comment]
Need to also handle the case where there's a node failure and `--no-kill` is used. In that case, a job that fails would be `RESIZING`, with another job that continues.

**Comment by hagertnl** [comment]
Converted to draft -- need to optimize & add absolute time options (`--starttime`, `--endtime`)

**Comment by hagertnl** [comment]
TODO -- need to copy all the fields we can (runarchive, builddir, etc) from the most recent record, otherwise Flux queries will not be able to query correctly.

**Comment by hagertnl** [comment]
TODO: need better in-code documentation of why `GET-v1` is used, and not GET

**Comment by hagertnl** [comment]
Expanding the scope of this PR to include improvements to other utilities as well.

**Review by ddietz89 — APPROVED** [review]
LGTM

**Pull request merged**